playground/resolve/index.html

```diff
@@ -214,6 +221,14 @@ <h2>resolve package that contains # in path</h2>
   import { msg as exportsWithModule } from '@vitejs/test-resolve-exports-with-module'
   text('.exports-with-module', exportsWithModule)
 
+  import { msg as exportsWithModuleCondition } from '@vitejs/test-resolve-exports-with-module-condition'
+  import { msg as exportsWithModuleConditionRequired } from '@vitejs/test-resolve-exports-with-module-condition-required'
```

Comment:
This fails when running `pnpm run test-build` but it works with `cd playground/resolve && pnpm run dev` (thanks to adding `@vitejs/test-resolve-exports-with-module-condition-required` to `optimizeDeps` [here](https://github.com/vitejs/vite/pull/13370/files#diff-49424114637c1e82467b027bb072c5f4b31ba8c74f46ce8ffa48bf6ccb802562R105)). How should I configure the actual test script to do the right thing here (to process linked CJS modules)?

Reply:
If the dep needs to be optimized during build time, you could set `optimizeDeps.disabled: false` in the config (the default is disabled 'build'). But I think we may need to move these tests to the [optimize-deps playground](https://github.com/vitejs/vite/tree/main/playground/optimize-deps) though as we aren't yet testing deps optimization at build for all playgrounds (it is an experimental feature).

Reply:
What I don't understand though is the fact that there are some other CJS modules in `playground/resolve`. So it seems like should be possible to keep this test here (I feel like it belongs to this particular test suite, otoh - if that's problematic in the current setup... it's definitely not something worth fighting for). Do you happen to know what's the difference between those other tests and this one?

Reply:
Hm, so I don't really want to get those deps optimized anyhow (I think?), i'd only like to apply commonjs interop here. Or is the `'module'` condition handling part of the optimizations in Vite?

Reply:
It works fine during build if you use `index.cjs` instead of a `.js` file in the `@vitejs/test-resolve-exports-with-module-condition-required`. This is also how other CJS packages are constructed in the resolve playground. I thought yours should work too though.

Reply:
Thanks for the tip. I changed the extension and it seems to work now - the CI should pass shortly

Reply:
Another way is to use the `file:` protocol like `"file:./exports-with-module-condition-required"` in the `package.json` so it's hardlinked instead, but either way also works fine.